resip/recon/UserAgent.cxx

```diff
@@ -569,6 +610,79 @@ UserAgent::destroySubscriptionImpl(SubscriptionHandle handle)
    }
 }
 
+void 
+UserAgent::createPublicationImpl(PublicationHandle handle, const Data& status, const Data& eventType, const NameAddr& target, unsigned int publicationTime, const Mime& mimeType)
+{
+   // Ensure we have a client publication handler for this event type
+   if(!mDum.getClientPublicationHandler(eventType))
+   {
+      mDum.addClientPublicationHandler(eventType, this);
+   }
+   // Ensure that the request Mime type is supported in the dum profile
+   if(!mProfile->isMimeTypeSupported(PUBLISH, mimeType))
+   {
+      mProfile->addSupportedMimeType(PUBLISH, mimeType);  
```

Comment:
I not 100% sure, but I'm thinking it's not needed.

On Thu, Aug 11, 2016 at 3:22 AM, Daniel Pocock <notifications@github.com>
wrote:

> In resip/recon/UserAgent.cxx
> <https://github.com/resiprocate/resiprocate/pull/52#discussion_r74377380>:
>
> > @@ -569,6 +610,79 @@ UserAgent::destroySubscriptionImpl(SubscriptionHandle handle)
> >     }
> >  }
> >
> > +void
> > +UserAgent::createPublicationImpl(PublicationHandle handle, const Data& status, const Data& eventType, const NameAddr& target, unsigned int publicationTime, const Mime& mimeType)
> > +{
> > +   // Ensure we have a client publication handler for this event type
> > +   if(!mDum.getClientPublicationHandler(eventType))
> > +   {
> > +      mDum.addClientPublicationHandler(eventType, this);
> > +   }
> > +   // Ensure that the request Mime type is supported in the dum profile
> > +   if(!mProfile->isMimeTypeSupported(PUBLISH, mimeType))
> > +   {
> > +      mProfile->addSupportedMimeType(PUBLISH, mimeType);
>
> OK, in that case, you can leave it like that
>
> @sgodin <https://github.com/sgodin> should a PUBLISH client need to call
> addSupportedMimeType given that is is only sending messages?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/resiprocate/resiprocate/pull/52/files/e0897e2885abd5e59bea8b7669ce3c42e9295d93#r74377380>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACtNUdr9GcqWjZvlv6bfphqqHlqHE5fZks5qes2xgaJpZM4JYfsi>
> .
>